packages/client/src/hooks/useNotification.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useRef } from 'react';
+import shipIcon from '@assets/icons/ship.png';
+
+interface useNotificationOptions {
+	onClick?: () => void;
+}
+
+const title = 'Galaxy Conquerors';
+
+const createNotification = (text: string) => {
+	return new Notification(title, {
+		icon: shipIcon,
+		body: text
+	});
+};
+
+/**
+ * Хук для отображения уведомлений
+ * @param options Опции уведомления
+ * @returns Функция, которя принимает текст сообщения и вызывает уведомление
+ */
+export const useNotification = (options: useNotificationOptions) => {
+	const notification = useRef<Notification>();
+
+	const notify = useCallback(
+		(text: string) => {
+			if (!('Notification' in window)) {
+				return;
+			} else if (Notification.permission === 'granted') {
+				notification.current = createNotification(text);
+			} else if (Notification.permission !== 'denied') {
+				Notification.requestPermission().then(permission => {
+					if (permission === 'granted') {
+						notification.current = createNotification(text);
+					}
+				});
+			}
```

Comment:
_Не критично, но лучше исправить_

Тут получилось нагромождение из-за `else if`, можно исправить это и улучшить читаемость

После `return` `else` не нужен, так как код не дойдет до этого места в любом случае

После обработки `granted` предлагаю убрать проверку на `denied` 
`if (Notification.permission !== 'denied')`
и просто запрашивать разрешение для всех состояний кроме `granted`
```suggestion
			if (!('Notification' in window)) {
				return;
			} 
			if (Notification.permission === 'granted') {
				notification.current = createNotification(text);
			} else {
				Notification.requestPermission().then(permission => {
					if (permission === 'granted') {
						notification.current = createNotification(text);
					}
				});
			}
```

Reply:
В таком случае пользователь будет получать предложение получать уведомления каждый раз при входе в игру, даже если он уже отказался от уведомлений. Мне кажется это нехорошо. Оставлю пока логику такой.